code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -479,88 +479,263 @@
+			UnregisterSignal(mind_target, COMSIG_MIND_TRANSFER_TO)
+		if(mind_target.current)
+			UnregisterSignal(mind_target.current, list(COMSIG_MOB_LOGIN, COMSIG_MOB_DEATH))
+
+/**
+ * Ensures a mind is a valid candidate for being slime linked.
+ *
+ * Arguments
+ * * mind: The mind to validate for slime linking.
+ * * initial_connection: Whether or not this is the first connection attempt (rather than validating to ensure a mind is still allowed to be in the slime link).
+ * * silent: Whether to display messages when validating or not.
+ * Returns TRUE if the mind is a valid candidate for being slime linked, FALSE otherwise.
+ */
+/datum/species/oozeling/stargazer/proc/validate_mind(datum/mind/mind, initial_connection = FALSE, silent = FALSE)
+	. = TRUE
+	if(QDELETED(mind?.current) || mind.current.stat == DEAD)
 		return FALSE
-	if(HAS_TRAIT(M, TRAIT_MINDSHIELD)) //mindshield implant, no dice
+	if(initial_connection && GLOB.slime_links_by_mind[mind])
+		var/datum/weakref/other_link_ref = GLOB.slime_links_by_mind[mind]
+		var/datum/species/oozeling/stargazer/other_link = other_link_ref?.resolve()
+		// If they're already slime linked, then we can't link to them.
+		if(other_link && istype(other_link))
+			return FALSE
+	if(istype(mind.current.get_item_by_slot(ITEM_SLOT_HEAD), /obj/item/clothing/head/foilhat))
+		if(!silent)
+			var/mob/living/carbon/human/owner = slimelink_owner.resolve()
+			if(owner)
+				to_chat(mind.current, "<span class='danger'>[owner]'s no-good syndicate mind-slime is blocked by your protective headgear!</span>")
+		return FALSE
+	if(HAS_TRAIT_NOT_FROM(mind.current, TRAIT_MINDSHIELD, "nanites")) //mindshield implant, no dice
 		return FALSE
-	var/mob/living/carbon/human/owner = slimelink_owner.resolve()
-	if(istype(M.get_item_by_slot(ITEM_SLOT_HEAD), /obj/item/clothing/head/foilhat))
-		if(owner)
-			to_chat(M, "<span class='danger'>[owner.real_name]'s no-good syndicate mind-slime is blocked by your protective headgear!</span>")
 
+/**
+ * Handles the death of a linked mob, unlinking them from the slime link.
+ *
+ * Arguments
+ * * source_mob: The mob that died.
+ */
+/datum/species/oozeling/stargazer/proc/link_death(mob/living/source_mob)
+	SIGNAL_HANDLER
+	if(!source_mob.mind || !(source_mob.mind in linked_minds))
+		return
+	var/datum/weakref/owner_ref = GLOB.slime_links_by_mind[source_mob.mind]
+	var/datum/species/oozeling/stargazer/stargazer = owner_ref?.resolve()
+	var/mob/living/carbon/human/link_owner = stargazer?.slimelink_owner?.resolve()
+	if(!link_owner || !istype(link_owner))
+		return
+	to_chat("<span class='danger'>As you die, you feel your link to [link_owner.real_name] fizzle out!</span>")
+	unlink_mind(source_mob.mind)
+
+/**
+ * Links the mind of another mob to this stargazer's slime link.
+ *
+ * Arguments
+ * * mind: The mind to link to the slime link.
+ * Returns TRUE if the mind was successfully linked, FALSE otherwise.
+ */
+/datum/species/oozeling/stargazer/proc/link_mind(datum/mind/mind)
```

Comment:
I don't like the var name is just `mind` here. I feel like this implies the slime link owner's mind, not target mind. `target_mind` would fit better.